### PR TITLE
Fixed error when reading out the data

### PR DESCRIPTION
**Pull request by steffi3011**
The variable "count" is now set correctly.

### CODE REVIEW

**Comment by steffi3011** [comment]
#80 Error reading the data

**Comment by ironfede** [comment]
Thank you @steffi3011  for your interest in OpenMcdf. I think that surely You've found a bug but not in the meaning implied by your pull request. Description of parameter `offset` is simply wrong (I'm going to fix it) because it's a refuse from some legacy version where it was publicly exposed by the api. Now, `offset` is the receiving buffer offset, not the stream one and when you calculate the number of bytes to read you have to choose the minimum between count and buffer dimension minus offset inside it. 
So, summing up, the correct way to do this should be
` count = (int)Math.Min((long)(buffer.Length - offset), (long)count);`
I'm going to commit this change even if the fact that offset should not be reachable from the public api mitigates its impact.
I hope to have been clear enough in this explanation, anyhow many thanks to you for your advice.
Federico

**Review by ironfede — COMMENTED** [review]
Thank you @steffi3011 for your interest in OpenMcdf. I think that surely You've found a bug but not in the meaning implied by your pull request. Description of parameter offset is simply wrong (I'm going to fix it) because it's a refuse from some legacy version where it was publicly exposed by the api. Now, offset is the receiving buffer offset, not the stream one and when you calculate the number of bytes to read you have to choose the minimum between count and buffer dimension minus offset inside it.
So, summing up, the correct way to do this should be
count = (int)Math.Min((long)(buffer.Length - offset), (long)count);
I'm going to commit this change even if the fact that offset should not be reachable from the public api mitigates its impact.
I hope to have been clear enough in this explanation, anyhow many thanks to you for your advice.
Federico
